### DISCUSSION TITLE
error: not spawned with IPC, when I use the Node API child_process.exec() to run the command `code-server --install-extension <extension id>` in my extension code

### DISCUSSION THREAD

**Discussion by Longfd** [coder/code-server#5806] [Q&A]
When I run the install command in the terminal, it runs well.
My questions is:
- Is it not allowed to call `code-server` inside vscode extension?

**Comment by Longfd** [comment]
Use the vscode terminal API can solve my problem, [example](https://github.com/microsoft/vscode-extension-samples/tree/main/extension-terminal-sample).
But I still don't know why the error 'not spawned with IPC'  happens

**Reply by jsjoeio** [reply]
Very interesting! @code-asher any ideas?

**Reply by code-asher** [reply]
Would definitely recommend using the API if possible.

For the `not spawned with IPC` error that is probably happening because of the `CODE_SERVER_PARENT_PID` environment variable.  We should unset that in the environment host.

Basically `CODE_SERVER_PARENT_PID` makes code-server think it is the inner process so it tries to communicate with the parent via IPC but there is no parent and therefore no IPC.
